tests/ui/as_slice_instead_of_reference_full_range.rs

```diff
@@ -0,0 +1,20 @@
+#![warn(clippy::as_slice_instead_of_reference_full_range)]
+
+mod nested_1 {
+    pub(crate) mod nested_2 {
+        pub(crate) const SOME_VALUE: [u8; 4] = [1, 2, 3, 4];
+    }
+}
+
+fn main() {
+    let array: [u8; 4] = [0; 4];
+    let slice: &[u8] = &array[..];
+    //~^ as_slice_instead_of_reference_full_range
+
+    let mut array: [u8; 4] = [0; 4];
+    let mut slice: &[u8] = &mut array[..];
+    //~^ as_slice_instead_of_reference_full_range
+
+    let slice: &[u8] = &nested_1::nested_2::SOME_VALUE[..];
+    //~^ as_slice_instead_of_reference_full_range
```

Comment:
I'd like to see some tests where the array and the slice come from different expansion contexts (e.g. a `macro_rules!` that does the slicing to an argument).

Also a test where we slice something that is already a slice instead of an array, e.g. `let slice: &[u8] = b"foo"; &foo[..]`.